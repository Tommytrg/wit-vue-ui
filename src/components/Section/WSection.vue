<template>
  <div :class="`w-full h-full ${frameClasses}`">
    <div
      :class="`content p-[100px] m-[0_auto] max-w-[1100px] w-full sm:p-[70px_32px] xs:p-[70px_16px] z-50 ${contentClasses}`"
    >
      <slot name="content"></slot>
    </div>
  </div>
</template>

<script setup lang="ts">
defineProps({
  frameClasses: {
    type: String,
    default: ''
  },
  contentClasses: {
    type: String,
    default: ''
  }
})
</script>
