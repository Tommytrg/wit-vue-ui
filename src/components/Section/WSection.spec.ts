import { describe, it, expect } from 'vitest'
import { h } from 'vue'
import { mount } from '@vue/test-utils'
import WSection from './WSection.vue'

describe('WSection', () => {
  it('renders properly', () => {
    const wrapper = mount(WSection, {
      props: {
        frameClasses: 'bg-black-950',
        contentClasses: 'grid justify-items-center',
      },
      slots: {
        content: h('p', 'Section content')
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})