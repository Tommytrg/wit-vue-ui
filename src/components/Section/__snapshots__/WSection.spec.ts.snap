// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WSection > renders properly 1`] = `
<div
  class="w-full h-full bg-black-950"
>
  <div
    class="content p-[100px] m-[0_auto] max-w-[1100px] w-full sm:p-[70px_32px] xs:p-[70px_16px] z-50 grid justify-items-center"
  >
    
    <p>
      Section content
    </p>
    
  </div>
</div>
`;
